delphi/translators/for2py/rectify.py

```diff
@@ -0,0 +1,1982 @@
+
+Example:
+    This script is executed by the autoTranslate script as one
+    of the steps in converted a Fortran source file to Python
+    file. For standalone execution:::
+
+        $python rectify.py <ast_file>
+
+ast_file: The XML represenatation of the AST of the Fortran file. This is
+produced by the OpenFortranParser.
+
+Author: Terrence J. Lim
+Last Modified: 4/11/2019
+"""
+
+import sys
+import re
+import argparse
+import xml.etree.ElementTree as ET
+
+
+class RectifyOFPXML:
+    def __init__(self):
+        self.is_derived_type = False
+        self.is_derived_type_ref = False
+        self.is_array = False
+        self.is_format = False
+        self.is_function_arg = False
+        self.need_reconstruct = False
+
+        self.cur_derived_type_name = ""
+        self.current_scope = ""
+
+        # Keep a track of both array and non-array variables in the dictionary
+        # of {'name' : 'scope'}
+        self.declared_non_array_vars = {}
+        self.declared_array_vars = {}
+
+        self.derived_type_var_holder_list = []
+        self.derived_type_refs = []
+        self.subscripts_holder = []
+
+        self.format_holder = ET.Element("")
+        self.parent_type = ET.Element("")
+        self.derived_type_ref = ET.Element("")
+        self.current_body_scope = ET.Element("")
+
+    """
+        Nested child tag list
+    """
+    FILE_CHILD_TAGS = ["program", "subroutine", "module"]
+
+    STATEMENT_CHILD_TAGS = [
+        "assignment",
+        "write",
+        "format",
+        "stop",
+        "execution-part",
+        "print",
+        "open",
+        "read",
+        "close",
+        "call",
+        "statement",
+        "label",
+        "literal",
+        "continue-stmt",
+        "do-term-action-stmt",
+        "return",
+        "contains-stmt",
+        "declaration",
+        "prefix",
+        "function",
+        "internal-subprogram",
+        "internal-subprogram-part",
+        "prefix",
+        "exit",
+    ]
+
+    LOOP_CHILD_TAGS = ["header", "body", "format"]
+
+    DECLARATION_CHILD_TAGS = [
+        "type",
+        "dimensions",
+        "variables",
+        "format",
+        "name",
+    ]
+
+    DERIVED_TYPE_CHILD_TAGS = [
+        "declaration-type-spec",
+        "type-param-or-comp-def-stmt-list",
+        "component-decl-list__begin",
+        "component-initialization",
+        "data-component-def-stmt",
+        "component-def-stmt",
+        "component-attr-spec-list",
+        "component-attr-spec-list__begin",
+        "explicit-shape-spec-list__begin",
+        "explicit-shape-spec",
+        "explicit-shape-spec-list",
+        "component-attr-spec",
+        "component-attr-spec-list__begin",
+        "component-shape-spec-list__begin",
+        "explicit-shape-spec-list__begin",
+        "explicit-shape-spec",
+        "component-attr-spec",
+        "component-attr-spec-list",
+        "end-type-stmt",
+        "derived-type-def",
+    ]
+
+    """
```

Comment:
Can you move this docstring (and the other similar ones in the rest of the module) into the function? This way they will be automatically associated with the function and show up in the autogenerated Sphinx documentation.

See https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google for examples.